File: 492. Construct the Rectangle/492.cpp
class Solution {
public:
    vector<int> constructRectangle(int area) {
        vector<int> ans;
        int L = area;
        for (int i=int(sqrt(area)); i>0; --i){
            if (!(area%i)) {
                L = (i>area/i) ? i : area/i;
                break;
            }
        }
        ans.push_back(L), ans.push_back(area/L);
        return ans;
    }
};

File: 492. Construct the Rectangle/492.py
class Solution(object):

    def constructRectangle(self, area):
        """
        :type area: int
        :rtype: List[int]
        """
        for i in range(int(math.sqrt(area)), 0, -1):
            if not area % i:
                return [i, int(area / i)] if i > int(area / i) else [int(area / i), i]
